eox_nelp/api_clients/certificates.py

```diff
@@ -0,0 +1,68 @@
+"""Client module for external certificate's API integration.
+
+Classes:
+    ExternalCertificatesApiClient: Class to interact with NELP external certificates service.
+"""
+from django.conf import settings
+
+from eox_nelp.api_clients import AbstractApiClient
+
+
+class ExternalCertificatesApiClient(AbstractApiClient):
+    """Allow to perform multiple external certificates operations."""
+
+    def __init__(self):
+        client_id = getattr(settings, "EXTERNAL_CERTIFICATES_API_CLIENT_ID")
+        client_secret = getattr(settings, "EXTERNAL_CERTIFICATES_API_CLIENT_SECRET")
+
+        super().__init__(client_id, client_secret)
+
+    @property
+    def base_url(self):
+        return getattr(settings, "EXTERNAL_CERTIFICATES_API_URL")
+
+    def create_external_certificate(self, certificate_data):
+        """This will create an external certificate based on the input data, this data should have the
+        following keys:
+
+            id <mandatory>: edx-platform certificate identifier.
+            created_at <mandatory>: when the certificate was created.
+            expiration_date <mandatory>: when the certificate expires.
+            grade <mandatory>: The associated grade with the certificate.
+            is_passing <mandatory>: Boolean value that represent if the user has passed the course.
+            user <mandatory>: Dictionary with the following data:
+                national_id: User National identifier.
+                englishs_name <optional>: User name in English.
+                arabic_name <optional>: User name in Arabic.
+        Args:
+            certificate_data<Dict>: Information about a the certificate.
+
+        Returns:
+            response<Dict>: requests response as dictionary.
+
+        Raise:
+            KeyError: This will be raised when the mandatory are excluded in the certificate data.
+        """
+        path = "certificates"  # This is not clear at all
+        user = certificate_data["user"]
+        payload = {
+            "reference_id": certificate_data["id"],
+            "date": {
+                "issuance": str(certificate_data["created_at"]),
+                "expiration": str(certificate_data["expiration_date"]),
+            },
+            "individual": {
+                "name_en": user.get("english_name", ""),
```

Comment:
I have a question here.
In this section user is a dict and not a User object?

Reply:
https://github.com/eduNEXT/eox-nelp/blob/d12faa2b85365693fe1933d0830b090cca868dd5/eox_nelp/api_clients/certificates.py#L33

Reply:
Is strange, mainly time we manage we use the User Object,

Reply:
this is a new function therefore we have to define the signature, in this case there is only one input data and that is a dictionary, so imo to include models inside the dictionary will be strange since it is not an standard  python type  

Indeed here  my concern  was where should I include the transformation data, since the original data is a  [CertificateData](https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/data.py#L100) and we must send a different format(API format), please check this pr https://github.com/eduNEXT/eox-nelp/pull/76, and let me know your thoughts  about the general implementation

Reply:
That has total logic due to delegating async tasks we cant send objects in args, but we can send dicts.
So is like serializing the user object into dict.
https://github.com/eduNEXT/eox-nelp/pull/76/files#diff-902ce53d9739afb0b3d31006e61041cc409b76c1a250609145a838a8d070bf71R192